Extensions/xkit_patches.js

```diff
@@ -227,13 +227,51 @@ XKit.tools.make_gist = function(text, name) {
 			public: false,
 			files: files,
 		}),
-		// '{"description": "a gist for a user with token api call via ajax","public": false,"files": {"file1.txt": {"content": "testing 1 2 3"}}}'
 	}).then(function(resp){
 		return resp.html_url;
 	});
 };
 
 /**
+ * Creates a link to a github issue with error text and template
+ * @param {String} title - the title of the github issue--should be unique and useful
+ * @param {Object?} data - Key-value pairs to list at the top of the issue.
+ * @param {Error?} error - An exception to serialize, if availible
+ * @return {String} The url to link the user to
+ */
+XKit.tools.github_issue = function(title, data, error) {
+	/*jshint -W069 */
+
+	if (!data) {
+		data = {};
+	}
+
+	data['User Agent'] = window.navigator.userAgent;
+	data['XKit Version'] = XKit.version;
+	data['Patches Version'] = XKit.installed.get("xkit_patches").version;
+	data['Extensions'] = XKit.installed.list().join(", ");
```

Comment:
@wolvan I considered that, but I think its a) more complicated code for b)
probably not enough effort. We can include individual extension versions in
the data if they're important to the error. (and probably should for ER....)

On Mon, Jun 27, 2016 at 10:37 AM Wolvan <notifications@github.com> wrote:

> In Extensions/xkit_patches.js
> <https://github.com/new-xkit/XKit/pull/1089#discussion_r68587397>:
>
> > + * @param {String} title - the title of the github issue--should be unique and useful
> > + * @param {Object?} data - Key-value pairs to list at the top of the issue.
> > + * @param {Error?} error - An exception to serialize, if availible
> > + * @return {String} The url to link the user to
> > + */
> > +XKit.tools.github_issue = function(title, data, error) {
> > +	/*jshint -W069 */
> > +
> > +	if (!data) {
> > +		data = {};
> > +	}
> > +
> > +	data['User Agent'] = window.navigator.userAgent;
> > +	data['XKit Version'] = XKit.version;
> > +	data['Patches Version'] = XKit.installed.get("xkit_patches").version;
> > +	data['Extensions'] = XKit.installed.list().join(", ");
>
> Shouldn't we include Extension versions here too?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/new-xkit/XKit/pull/1089/files/556891a4598952ac1b19183a8857fe17ddfccb3c#r68587397>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAORVxZ4v_oJ3JLnw7CvyCu6Bxl1CdO0ks5qP-AcgaJpZM4I-v_4>
> .
>